internal/config/config.go

```diff
@@ -319,9 +325,35 @@ func peerFromCR(p metallbv1beta1.BGPPeer) (*Peer, error) {
 		}
 	}
 
+	if p.Spec.Password != "" && p.Spec.SecretRef != nil {
```

Comment:
Can you add a unit test that verifies this fails?

Reply:
done, added relevant unit tests to cover this change.